jobs/src/main/scala/dcos/metronome/utils/state/EntityStoreCache.scala

```diff
@@ -0,0 +1,123 @@
+package dcos.metronome.utils.state
+
+import mesosphere.marathon.PrePostDriverCallback
+import mesosphere.marathon.state.VersionedEntry
+import org.slf4j.LoggerFactory
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.Future
+
+/**
+  * The entity store cache caches all current versions from the underlying store.
+  * During election we prefetch al unversioned entries.
+  *
+  * Idea:
+  * - old versions (versioned entries) are never cached
+  * - we always cache the current version
+  * - on elected:
+  *    - clear everything
+  *    - load all current versions
+  *  - on defeated:
+  *    - clear everything
+  */
+class EntityStoreCache[T <: MarathonState[_, T]](store: EntityStore[T])
```

Comment:
Do we need a cache?

Reply:
yep, I just pulled everything from 1.3 in here. I would be inclined not to touch it :)

Reply:
yeah... I'm hoping not to touch it either... it is temp until we move to the persistence marathon moved to in 1.4.